app/assets/javascripts/controllers/network_router/network_router_form_controller.js

```diff
@@ -1,33 +1,70 @@
+    enable_snat: true,
+    external_gateway: false,
+    extra_attributes: null,
   };
+
   vm.formId = networkRouterFormId;
-  vm.afterGet = false;
-  vm.modelCopy = angular.copy( $scope.networkRouterModel );
   vm.model = "networkRouterModel";
-
+  vm.newRecord = networkRouterFormId === "new";
   vm.saveable = miqService.saveable;
 
-  ManageIQ.angular.scope = vm;
-
-  if (networkRouterFormId == 'new') {
-    vm.networkRouterModel.name = "";
-    vm.networkRouterModel.enable_snat = true;
-    vm.networkRouterModel.external_gateway = false;
-    vm.networkRouterModel.cloud_subnet_id = null;
-    vm.newRecord = true;
+  if (vm.newRecord) {
+    vm.afterGet = true;
+    vm.modelCopy = angular.copy(vm.networkRouterModel);
   } else {
     miqService.sparkleOn();
-
-    $http.get('/network_router/network_router_form_fields/' + networkRouterFormId)
-      .then(getNetworkRouterFormData)
-      .catch(miqService.handleFailure);
+    API.get("/api/network_routers/" + networkRouterFormId + "?attributes=name,ems_id,admin_state_up,cloud_network_id,extra_attributes,cloud_tenant,ext_management_system,cloud_subnets").then(function(data) {
+      Object.assign(vm.networkRouterModel, data);
+      if (data.extra_attributes.external_gateway_info && data.extra_attributes.external_gateway_info !== {}) {
+        vm.networkRouterModel.external_gateway = true;
+      }
+    }).then(function() {
+      if (vm.networkRouterModel.external_gateway) {
+        getSubnetByRef(vm.networkRouterModel.extra_attributes.external_gateway_info.external_fixed_ips[0].subnet_id);
+      }
+    }).then(function() {
+      getCloudNetworksByEms(vm.networkRouterModel.ext_management_system.id);
+    }).then(function() {
+      getCloudSubnetsByNetworkID(vm.networkRouterModel.cloud_network_id);
+    }).then(function() {
+      vm.afterGet = true;
+      vm.modelCopy = angular.copy(vm.networkRouterModel);
+      miqService.sparkleOff();
+    }).catch(miqService.handleFailure);
   }
 
+  var getCloudNetworksByEms = function(id) {
+    if (id) {
+      API.get("/api/cloud_networks?expand=resources&attributes=name,ems_ref&filter[]=external_facing=true&filter[]=ems_id=" + id).then(function(data) {
+        vm.available_networks = data.resources;
+      }).catch(miqService.handleFailure);
+    }
+  };
+
+  var getCloudSubnetsByNetworkID = function(id) {
+    if (id) {
+      API.get("/api/cloud_subnets?expand=resources&attributes=name,ems_ref&filter[]=cloud_network_id=" + id).then(function(data) {
+        vm.available_subnets = data.resources;
+      }).catch(miqService.handleFailure);
+    }
+  };
+
+  var getSubnetByRef = function(ref) {
+    if (ref) {
+      API.get("/api/cloud_subnets?expand=resources&attributes=name&filter[]=ems_ref=" + ref).then(function(data) {
+        vm.networkRouterModel.cloud_subnet_id = data.resources[0].id;
+      }).catch(miqService.handleFailure);
+    }
+  };
+
   vm.addClicked = function() {
     var url = 'create/new' + '?button=add';
```

Comment:
This should be one string. No need for `+`. Could you please fix it? :)

Reply:
Done.